### PR TITLE
Add Azure support

### PR DESCRIPTION
**Pull request by AbelHu**

### CODE REVIEW

File: spec/bat/stemcell_spec.rb [hunk 1]
```diff
@@ -88,6 +88,9 @@
       'bosh-warden-boshlite-ubuntu-trusty-go_agent' => false,
       'bosh-warden-boshlite-centos-go_agent' => false,
 
+      # Azure CPI does not support network reconfig
```
Comment (cppforlife) [inline spec/bat/stemcell_spec.rb]: azure cpi does support network reconfig: https://github.com/AbelHu/bosh-azure-cpi-release/blob/develop/src/bosh_azure_cpi/lib/cloud/azure/cloud.rb#L188. it's just not efficient but the tests should pass anyway.

Comment (AbelHu) [inline spec/bat/stemcell_spec.rb]: As you said, we do not implement configure_networks. Next, we will add the support.

Comment (cppforlife) [inline spec/bat/stemcell_spec.rb]: You *did* implement it. `raise Bosh::Clouds::NotSupported` is a valid implementation. There is no need to change that implementation.

Comment (AbelHu) [inline spec/bat/stemcell_spec.rb]: I see. Will set it to true after I verify it. Thanks.

**Comment by cppforlife** [comment]
Entered two stories for us to investigate tests that you had to disable.

On your end, can you take advantage of following feature: https://github.com/cloudfoundry/bosh-acceptance-tests/pull/5/files and move the azure bats template file into bosh-azure-cpi-release. Also can you enable network configuration tests so that we dont have to pull in changes to spec/bat/stemcell_spec.rb.

**Comment by AbelHu** [comment]
I have update the latest BATs code into AbelHu/bosh-acceptance-tests.  Do you mean that I need to move azure.yml.erb (https://github.com/AbelHu/bosh-acceptance-tests/blob/master/templates/azure.yml.erb) to bosh-azure-cpi-release? Do you have any suggestion for where to put this file?

I enabled supports_network_reconfiguration for Azure last week. Is it not enough?
https://github.com/AbelHu/bosh-acceptance-tests/commit/2deb382690249af2f3603c2d63292f4b3493bf32

**Comment by cppforlife** [comment]
> Do you mean that I need to move azure.yml.erb to bosh-azure-cpi-release

Yup. Since you can now specify path to that file in https://github.com/Azure/bosh-azure-cpi-release/blob/master/ci/tasks/run-bats.sh#L49 it can live in cpi specific repo. We will be moving other templates as well.

> I enabled supports_network_reconfiguration for Azure last week. Is it not enough?

We are trying to make bats-acceptance-tests repo CPI agnostic so since that method does very specific thing for specific iaas-es there is no need to even add https://github.com/AbelHu/bosh-acceptance-tests/commit/2deb382690249af2f3603c2d63292f4b3493bf32. i.e. keep that method completely unmodified.

**Comment by AbelHu** [comment]
@cppforlife Update it.

**Comment by AbelHu** [comment]
After a discussion with @cppforlife, nothing change for Azure now so close it.

**Pull request closed without merging**